wz_strategic_plugin/include/wz_strategic_plugin/wz_strategic_plugin.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+/*
+ * Copyright (C) 2019-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <ros/ros.h>
+#include <string>
+#include <carma_utils/CARMAUtils.h>
+#include <cav_srvs/PlanManeuvers.h>
+#include <cav_msgs/Plugin.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <geometry_msgs/TwistStamped.h>
+#include <carma_wm/WMListener.h>
+#include <carma_wm/WorldModel.h>
+#include <carma_utils/CARMAUtils.h>
+#include <carma_wm/Geometry.h>
+
+#include <lanelet2_extension/regulatory_elements/CarmaTrafficLight.h>
+
+enum class CarmaTrafficLightState {UNAVAILABLE=0,DARK=1,STOP_THEN_PROCEED=2,STOP_AND_REMAIN=3,PRE_MOVEMENT=4,PERMISSIVE_MOVEMENT_ALLOWED=5,PROTECTED_MOVEMENT_ALLOWED=6,PERMISSIVE_CLEARANCE=7,PROTECTED_CLEARANCE=8,CAUTION_CONFLICTING_TRAFFIC=9};
+
+namespace wz_strategic_plugin
+{
+    class WzStrategicPlugin
+    {
+        public:
+            /**
+             * \brief Default constructor for WzStrategicPlugin class
+             */
+            WzStrategicPlugin() = default;
+
+            /**
+             * \brief General entry point to begin the operation of this class
+             */
+            void run();
+
+            /**
+             * \brief Initialize ROS publishers, subscribers, service servers and service clients
+             */
+            void initialize();
+
+
+            bool planManeuverCb(cav_srvs::PlanManeuversRequest &req, cav_srvs::PlanManeuversResponse &resp);
+
+            ros::Publisher plugin_discovery_pub_;
```

Comment:
Can the class members defined from Lines 58 to 82 to private? Or are they being called elsewhere